src/nginx_config.py

```diff
@@ -293,3 +297,12 @@ def _server(
                 *self._locations(addresses_by_role),
```

Comment:
We also need to add `self._resolver(...` here to cater for the non-TLS scenario.